tests/unit/s2n_self_talk_inet_socket_test.c

```diff
@@ -0,0 +1,163 @@
+
+    /* Setup config */
+    RESULT_GUARD_POSIX(s2n_connection_set_blinding(client_conn, S2N_SELF_SERVICE_BLINDING));
+    RESULT_GUARD_POSIX(s2n_connection_set_fd(client_conn, fd));
+    RESULT_GUARD_POSIX(s2n_config_set_cipher_preferences(config, "default"));
+    RESULT_GUARD_POSIX(s2n_config_set_unsafe_for_testing(config));
+    RESULT_GUARD_POSIX(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+    RESULT_GUARD_POSIX(s2n_connection_set_config(client_conn, config));
+
+    /* Do handshake */
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    RESULT_GUARD_POSIX(s2n_negotiate(client_conn, &blocked));
+    RESULT_ENSURE_EQ(client_conn->actual_protocol_version, S2N_TLS12);
+
+    char sync = 0;
+    char recv_buffer[1] = { 0 };
+
+    {
+        RESULT_GUARD_POSIX(read(read_pipe, &sync, 1));
+        RESULT_GUARD_POSIX(s2n_recv(client_conn, recv_buffer, 1, &blocked));
+        RESULT_ENSURE_EQ(memcmp(&CHAR_A, &recv_buffer[0], 1), 0);
+
+        RESULT_GUARD_POSIX(read(read_pipe, &sync, 1));
+        RESULT_GUARD_POSIX(s2n_recv(client_conn, recv_buffer, 1, &blocked));
+        RESULT_ENSURE_EQ(memcmp(&CHAR_B, &recv_buffer[0], 1), 0);
+    }
+
+    return S2N_RESULT_OK;
+}
+
+static S2N_RESULT start_server(int fd, int write_pipe)
+{
+    /* Setup connections */
+    DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+            s2n_connection_ptr_free);
+    DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+
+    DEFER_CLEANUP(struct s2n_cert_chain_and_key * chain_and_key, s2n_cert_chain_and_key_ptr_free);
+    EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+            S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
+
+    /* Setup config */
+    EXPECT_SUCCESS(s2n_connection_set_blinding(server_conn, S2N_SELF_SERVICE_BLINDING));
+    EXPECT_EQUAL(s2n_connection_get_delay(server_conn), 0);
+    EXPECT_SUCCESS(s2n_connection_set_fd(server_conn, fd));
+    EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default"));
+    EXPECT_SUCCESS(s2n_config_set_unsafe_for_testing(config));
+    EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+    EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+    /* Do handshake */
+    s2n_blocked_status blocked = S2N_NOT_BLOCKED;
+    EXPECT_SUCCESS(s2n_negotiate(server_conn, &blocked));
+    EXPECT_EQUAL(server_conn->actual_protocol_version, S2N_TLS12);
+
+    char sync = 0;
+    char send_buffer[1] = { 0 };
+    {
+        send_buffer[0] = CHAR_A;
+        EXPECT_SUCCESS(s2n_send(server_conn, send_buffer, 1, &blocked));
+        EXPECT_SUCCESS(write(write_pipe, &sync, 1));
+
+        send_buffer[0] = CHAR_B;
+        EXPECT_SUCCESS(s2n_send(server_conn, send_buffer, 1, &blocked));
+        EXPECT_SUCCESS(write(write_pipe, &sync, 1));
+    }
+
+    return S2N_RESULT_OK;
+}
+
+int main(int argc, char **argv)
```

Comment:
I'm not exactly sure what you're testing in this test. You seem to be testing that you can negotiate and send/recv data on a special socket? But I'm not sure why that needs to be tested. Maybe you could add a comment explaining the purpose of this test, as it's not obvious to me.
Are we planning on adding more to this test in the future?

Reply:
I mention the reason for this in the PR description. I dont want to add that in the code because this test really has nothing to do with kTLS (it just established the base case).

Adding comments about kTLS is confusing because we could add tests similar to this that dont deal with kTLS. The only special thing about this test is that it uses a `AF_INET` socket. Thats actually huge because thats what our customers use.

Reply:
If the test has nothing to do with kTLS, then we should have a description describing what it's testing without mentioning kTLS.

Reply:
Yeah, to me it doesn't really look like you're testing s2n here. It looks like you're testing the INET domain, which isn't really related to s2n. Is the point of this test to get a technique/method in place so that you _can_ test KTLS in the future?